core/src/databases/database.rs

```diff
@@ -83,16 +85,216 @@ impl Database {
                 .into_iter()
                 .collect::<Vec<_>>();
 
-                Ok(DatabaseSchema(
-                    rows.into_par_iter()
-                        .map(|(table, rows)| {
-                            Ok((
-                                table.table_id().to_string(),
-                                DatabaseSchemaTable::new(table, TableSchema::from_rows(&rows)?),
-                            ))
-                        })
-                        .collect::<Result<HashMap<_, _>>>()?,
-                ))
+                let schema = rows
+                    .par_iter()
+                    .map(|(table, r)| {
+                        Ok((
+                            table.table_id().to_string(),
+                            DatabaseSchemaTable::new(table.clone(), TableSchema::from_rows(&r)?),
+                        ))
+                    })
+                    .collect::<Result<HashMap<_, _>>>()?;
+
+                let returned_rows = match return_rows {
+                    true => Some(
+                        rows.into_iter()
+                            .map(|(table, rows)| (table.table_id().to_string(), rows))
+                            .collect::<HashMap<_, _>>(),
+                    ),
+                    false => None,
+                };
+
+                Ok((DatabaseSchema(schema), returned_rows))
+            }
+        }
+    }
+
+    pub async fn create_in_memory_sqlite_conn(
+        &self,
+        project: &Project,
+        store: Box<dyn Store + Sync + Send>,
+    ) -> Result<Connection> {
+        match self.db_type {
+            DatabaseType::REMOTE => Err(anyhow!(
+                "Cannot build an in-memory SQLite DB for a remote database."
+            )),
+            DatabaseType::LOCAL => {
+                let time_build_db_start = utils::now();
+                let (schema, rows_by_table) = self.get_schema(project, store.clone(), true).await?;
+                let rows_by_table = match rows_by_table {
+                    Some(rows) => rows,
+                    None => return Err(anyhow!("No rows found")),
+                };
+                utils::done(&format!(
+                    "DSSTRUCTSTAT Finished retrieving schema: duration={}ms",
+                    utils::now() - time_build_db_start
+                ));
+
+                let table_schemas: HashMap<String, TableSchema> = schema
```

Comment:
This type is exactly schema this variable should not exist

Reply:
Indeed this is not needed thanks

Reply:
done